src/main/java/sirius/web/security/UserContext.java

```diff
@@ -218,7 +218,12 @@ private void bindUserToRequest(WebContext ctx) {
         }
 
         if (currentUser != null) {
-            return Optional.of(currentUser);
+            if (currentUser.isLoggedIn()) {
+                setCurrentUser(currentUser);
+                return Optional.of(currentUser);
+            } else {
+                return Optional.empty();
+            }
```

Comment:
this _could_ be simplified to
```suggestion
            return Optional.of(currentUser).filter(UserInfo::isLoggedIn);
```